api/app/Rules/SkillNotUsedByActivePool.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Rules;
+
+use App\Models\Pool;
+use Carbon\Carbon;
+use Closure;
+use Database\Helpers\ApiErrorEnums;
+use Illuminate\Contracts\Validation\ValidationRule;
+
+class SkillNotUsedByActivePool implements ValidationRule
+{
+    /**
+     * Run the validation rule.
+     */
+    public function validate(string $attribute, mixed $value, Closure $fail): void
+    {
+        // validation fails if skill is in use by an active pool
+        $skillId = $value;
+        $activePools = Pool::where('published_at', '<=', Carbon::now()->toDateTimeString())
+            ->where('closing_date', '>', Carbon::now()->toDateTimeString())
+            ->get()
+            ->load(['essentialSkills', 'nonessentialSkills']);
+
+        foreach ($activePools as $pool) {
```

Comment:
I think you could do this faster by doing this as part of the query itself. Include a where clause which checks that the pool has `$skillId` in any of its essential skills or nonessential skills, and then just check if $activePools returned anything or not.

Reply:
Yeah, I guess we can do that and condense the rule a bit. 
Done in https://github.com/GCTC-NTGC/gc-digital-talent/pull/7762/commits/29d18f51a0eb4aa0e66deb2fc22644ea4cbae9fb